src/app/common/hooks/use-key-actions.ts

```diff
@@ -22,7 +22,12 @@ export function useKeyActions() {
       },
 
       generateWalletKey() {
-        return dispatch(keyActions.generateWalletKey());
+        const secretKey = generateSecretKey(256);
```

Comment:
This is nice, by pulling the secretKey generation out of the reducers for generating a wallet you can now have the same simple logic for generateWalletKey and unlockWallet